app/controllers/master/service_discovery/auth_controller.rb

```diff
@@ -11,7 +11,7 @@ def show
   protected
 
   def self_domain_url(self_domain)
-    options = params.except(:self_domain, :scope).merge(host: self_domain, controller: 'provider/admin/service_discovery/auth')
+    options = params.permit(:code, :referrer, :state).merge(host: self_domain, controller: 'provider/admin/service_discovery/auth', action: :show)
```

Comment:
```suggestion
    options = params.permit(:code, :referrer, :state).merge(host: self_domain, controller: controller_path, action: :show)
```

Reply:
⚠️  Nothing to do with this PR but the `self_domain` is suggested to a security issue. It is fetched from the request params.

Reply:
ack, will change, thanks